### PR TITLE
added check for invalid range in contour_plot and derivatives

### PR DESCRIPTION
**Pull request by jnash10**
### 📚 Description

Solves #35032 

Add if condition in misc.py which contour_plot uses to check for invalid range entered by user(xmin > xmax or ymin>ymax)

### 📝 Checklist

- [x ] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
@tobiasdiez Would you approve the workflows if you have the permission?

File: src/sage/plot/misc.py
Comment (kwankyu) [inline src/sage/plot/misc.py]: needs a space `# check` I would write `# check for valid range`

File: src/sage/plot/misc.py
Comment (kwankyu) [inline src/sage/plot/misc.py:149]: needs spaces around `>`. It seems you can use `[0]`, `[1]` instead of `[-2]` and `[-1]` respectively since ranges is normalized by now. I am not sure though.

Comment (jnash10) [reply src/sage/plot/misc.py:149]: Spaces added.

Have to use the minus method as range can be entered as (xmin, xmax) or (x,xmin,xmax) last I checked.

**Comment by kwankyu** [comment]
Look into the Build & Test check. There is a failed doctest.

**Comment by jnash10** [comment]
I think I found the error. One of the examples in the doctest is the following:
https://github.com/sagemath/sage/blob/05329f6e39dd7c214477bdb73c66b305f6870631/src/sage/plot/misc.py#L107-L111

Which has the range in the wrong format as in both examples, xmin=1 and xmax=-1 which is against the definition of range :
https://github.com/sagemath/sage/blob/05329f6e39dd7c214477bdb73c66b305f6870631/src/sage/plot/misc.py#L34-L36

Should I change the doctest example in this case and make it go from -1 to 1?

This test is failling the code edit as this is the very error it aims to catch.

File: src/sage/plot/misc.py
Comment (jnash10) [reply src/sage/plot/misc.py]: > Look into the Build & Test check. There are lots of failed doctests. It seems that `ranges[1]` may not exist.

Solved this. Some functions take range input differently, so nested my if condition inside another which checks if:
```
len(ranges) > 1
```
Haven't pushed it yet due to the issue mentioned in the comment below.

**Comment by kwankyu** [comment]
It seems we could just swap the two values to fix the error, and use the doctests to test this conveniency.

**Comment by jnash10** [comment]
> It seems we could just swap the two values to fix the error, and use the doctests to test this conveniency.

Nice idea!

**Comment by jnash10** [comment]
> It seems we could just swap the two values to fix the error, and use the doctests to test this conveniency.

Upon implementing it, the plot comes out okay:
image

Yet the doctest fails as the intermediate function is also apparently returning a wrapper:
```
File "src/sage/plot/misc.py", line 110, in sage.plot.misc.?
Failed example:
    sage.plot.misc.setup_for_eval_on_grid(x+y, [(y,1,-1),(x,-1,1)], return_vars=True)
Expected:
    (<sage...>, [(1.0, -1.0, 2.0), (-1.0, 1.0, 2.0)], [y, x])
Got:
    (<sage.plot.misc.FastCallablePlotWrapper object at 0x7f2dc08daa10>,
     [(-1.0, 1.0, 2.0), (-1.0, 1.0, 2.0)],
     [y, x])
```


This was the implementation:
```
    # check for invalid range entered by user (xmin > xmax or ymin > ymax) and swap the values if True
    if len(ranges) > 1:
        for i in range(len(ranges)):
            if (ranges[i][-2] > ranges[i][-1]):
                ranges[i] = list(ranges[i])
                ranges[i][-1], ranges[i][-2] = ranges[i][-2], ranges[i][-1]
                ranges[i] = tuple(ranges[i])
               
```

Converting tuple to list to tuple was necessary for the swap( as tuples are immutable). But after the conversion since the data type is same, the response should also ideally be the same.

**Comment by kwankyu** [comment]
You can just fix the minor changes in the return values in the doctests.

**Comment by kwankyu** [comment]
Remove unnecessary parentheses in the condition of `if` statement.

**Comment by jnash10** [comment]
> 

Upon making this change, all tests are passed for misc.py
```
./sage -t ...../misc.py
````
giving me:
```
All tests passed!
----------------------------------------------------------------------
Total time for all tests: 4.0 seconds
    cpu time: 3.3 seconds
    cumulative wall time: 4.0 seconds
```

Is that enough or should I now run:
```
./sage -t -all
```
?

**Comment by kwankyu** [comment]
Normally, you don't need to because workflows for checking the branch should be run by now. But, as you are first-time contributor, it seems that running workflows require admin's permission.  @tobiasdiez may take care of you.

**Comment by kwankyu** [comment]
@mkoeppe @dimpase @roed314 

Could triage team have the permission to approve to run workflows for first-time contributors?

**Comment by roed314** [comment]
@kwankyu Unfortunately, we can't change the [permissions](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization) on a fine-grained level without paying for Github's enterprise level.  But I'll approve the run here.

**Comment by kwankyu** [comment]
Thanks. I wonder if we can run a bot approving workflows automatically for PRs with "secure" code or with certain label...

**Comment by roed314** [comment]
I don't know how we would check that code is secure....

**Comment by kwankyu** [comment]
We may consider code is  basically secure when there's no change in github workflows. Other than that, the reviewer should decide whether to trust the contributor and add a label to signify that. But I don't know the technical side whether github allows such a bot. Logically, it seems, the bot should not be a github workflow.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by kwankyu** [comment]
> Is there anything else I need to do for this particular PR/issue? (squash commits etc.? 

Once you get "positive review", you are all set.

Anyway, for this PR, if you are willing, please make the comment `# check ...` short  (according to PEP8). You can also squash commits.

> and in that case, do the checks happen again?

Yes. But never mind that.

File: src/sage/plot/misc.py
Comment (jnash10) [inline src/sage/plot/misc.py]: Updating comment to be shorter.

**Comment by jnash10** [comment]
> > Is there anything else I need to do for this particular PR/issue? (squash commits etc.?
> 
> Once you get "positive review", you are all set.
> 
> Anyway, for this PR, if you are willing, please make the comment `# check ...` short (according to PEP8). You can also squash commits.
> 
> > and in that case, do the checks happen again?
> 
> Yes (but no for this PR). But don't mind that.

Thank you.

Comment updated!

**Review by kwankyu — APPROVED** [review]
Looks good. Thank you.

**Comment by jnash10** [comment]
Removed trailing whitespace in the comment.

**Review by kwankyu — APPROVED** [review]
Okay

**Pull request merged**
